lib/saluki-core/src/runtime/mod.rs

```diff
@@ -0,0 +1,65 @@
+//! Runtime system.
+//!
+//! This module contains the core components of the runtime system, including supervisors and processes. It is directly
+//! inspired by [Erlang/OTP](https://www.erlang.org/docs/28/system/design_principles#supervision-trees).
+//!
+//! To quote the Erlang/OTP documentation:
+//!
+//! > Workers are processes that perform computations and other actual work. Supervisors are processes that monitor
+//! > workers. A supervisor can restart a worker if something goes wrong. The supervision tree is a hierarchical
+//! > arrangement of code into supervisors and workers, which makes it possible to design and program fault-tolerant
+//! > software.
+//!
+//! # Processes
+//!
+//! An asynchronous system is composed of independent units of computation running concurrently, such as a set of tasks
+//! executing on a thread pool. We refer to these as **processes.** In other systems, these might be called _actors_,
+//! _tasks_, _fibers_, _virtual threads_, _goroutines_, or something else. Processes are lightweight and able to be
+//! (generally) created and destroyed cheaply.
+//!
+//! Processes have a few key attributes and invariants:
+//!
+//! - every process is a future that runs as an independent asynchronous task on a Tokio runtime
+//! - every process has a unique numerical identifier and a semi-unique name
+//!
+//! Processes cannot run by themselves, however. They must be _supervised_.
```

Comment:
Confirming, unlike Erlang processes saluki-core processes do not have a mailbox / input queue associated for message passing?

Reply:
Confirmed later, no mailbox. Probably worth adding this to the documentation in case someone -- like myself -- comes along with an Erlang mental model inappropriately.